parallel_statistics/mean_variance.py

```diff
@@ -276,23 +330,26 @@ def run(self, iterator, comm=None, mode="gather"):
             self.add_data(*values)
         return self.collect(comm=comm, mode=mode)
 
-
-    def _accumulate(self, weight, mean, sq, w, m, s):
+    @staticmethod
+    def _accumulate_sparse(weight, mean, sq, w, m, s):
         # Algorithm from Shubert and Gertz.
-        if self.sparse:
-            weight = weight + w
-            delta = m - mean
-            mean = mean + (w / weight) * delta
-            delta2 = m - mean
-            sq = sq + s + w * delta * delta2
-        else:
-            good = w!=0
-            weight[good] = weight[good] + w[good]
-            delta = m[good] - mean[good]
-            mean[good] = mean[good] + (w[good] / weight[good]) * delta
-            delta2 = m[good] - mean[good]
-            sq[good] = sq[good] + s[good] + w[good] * delta * delta2
+        weight = weight + w
+        delta = m - mean
+        mean = mean + (w / weight) * delta
+        delta2 = m - mean
+        sq = sq + s + w * delta * delta2
 
         return weight, mean, sq
 
+    @staticmethod
+    @jit
+    def _accumulate_dense(weight, mean, sq, w, m, s):
```

Comment:
this function won't be faster as a numba jitted function the way it is writting.  It will only have benefit if it is turned into a loop.   I think the main benefit would be not creating all those temporary arrays which could be significant, but I would profile if you haven't already.

Reply:
Thanks - this was a bit of an afterthought as it isn't the slow bit.  Given this, I've undone the jitting here, since it won't be any real gain anyway.